StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet.swift

```diff
@@ -13,6 +13,75 @@ import PassKit
 @_spi(STP) import StripePaymentsUI
 import UIKit
 
+protocol IntentAbstraction: PaymentMethodRequirementProvider {
+    /// Extracts all the recommended `PaymentMethodType`s from the given `intent`.
+    /// - Returns: An ordered list of all the `PaymentMethodType`s for this `intent`.
+    var recommendedPaymentSheetMethodTypes: [PaymentSheet.PaymentMethodType] { get }
+    
+    var recommendedPaymentMethodTypes: [STPPaymentMethodType] { get }
```

Comment:
is this another luxe two-sources-of-truth thing?

Reply:
Yes 🙃